var/spack/repos/spack_repo/builtin/build_systems/meson.py

```diff
@@ -224,3 +241,101 @@ def check(self) -> None:
         with working_dir(self.build_directory):
             self.pkg._if_ninja_target_execute("test")
             self.pkg._if_ninja_target_execute("check")
+
+
+def define(meson_var: str, value: Any) -> str:
+    """Return a meson command line argument that defines a variable.
+
+    The resulting argument will convert boolean values to false/true
+    and lists/tuples to meson colon-separated string lists. All other
+    values will be interpreted as strings.
+
+    Examples:
+
+        .. code-block:: python
+
+            [define('prefer_static', True),
+             define('cpp_std', 14),
+             define('swr', ['avx', 'avx2'])]
+
+        will generate the following configuration options:
+
+        .. code-block:: console
+
+            ["-Dprefer_static=true",
+             "-Dcpp_std=14",
+             "-Dswr=avx,avx2]
+
+    """
+    # Create a list of pairs. Each pair includes a configuration
+    # option and whether or not that option is activated
+    if isinstance(value, bool):
+        value = "true" if value else "false"
+    else:
+        if isinstance(value, collections.abc.Sequence) and not isinstance(value, str):
+            value = ",".join(str(v) for v in value)
```

Comment:
If `value` contains a comma character, which is admittedly uncommon but is a legal ASCII string and may be used for obscure things for who knows what reasons...

This becomes ambiguous.

Meson accepts two formats for array/combo options:

```
"-Doptname=val1,val2,val3"

"-Doptname=['val1', 'val2', 'val3']"
```

Up to you, which one you use. :) The former is more pleasant to read and write by hand, if it works.

Related: https://gitweb.gentoo.org/repo/gentoo.git/tree/eclass/meson.eclass#n87
https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-build/meson-format-array